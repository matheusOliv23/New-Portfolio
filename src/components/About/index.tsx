import React from "react"
import { motion } from "framer-motion"
import tech from "src/lotties/tech.json"
import Lottie from "react-lottie"
import { useTranslation } from "react-i18next"

export default function About() {
  const { t } = useTranslation()
  const about = [
    t("about.skills"),
    t("about.proficient"),
    t("about.proactive"),
    t("about.always")
  ]

  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: tech,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  }

  return (
    <section
      id="about"
      className="flex w-full flex-col bg-gray-light font-openSans gap-10 px-7 py-6 md:py-20 lg:py-32 m-auto"
    >
      <motion.div
        initial="hidden"
        whileInView="visible"
        variants={{
          visible: { opacity: 1, scale: 1 },
          hidden: { opacity: 0, scale: 0 }
        }}
        transition={{
          duration: 0.6,
          delay: 0.5,
          ease: [0, 0.71, 0.2, 1.01]
        }}
      >
        <h2 className="text-[40px] text-center">Sobre Mim</h2>
      </motion.div>
      <div className="lg:flex-row lg:gap-6 items-center flex m-auto flex-col max-w-7xl xl:justify-between lg:justify-between">
        <div className="space-y-4">
          <motion.div
            initial="hidden"
            whileInView="visible"
            transition={{ duration: 0.3 }}
            variants={{
              visible: { opacity: 1, scale: 1 },
              hidden: { opacity: 0, scale: 0 }
            }}
            className="lg:max-w-xl text-xl text-justify"
          ></motion.div>
          <ul className="!list-disc space-y-4 md:pl-6">
            {about.map((item, index) => (
              <li key={index} className="list-disc">
                <motion.div
                  initial="hidden"
                  whileInView="visible"
                  transition={{ duration: 0.8 }}
                  variants={{
                    visible: { opacity: 1, scale: 1 },
                    hidden: { opacity: 0, scale: 0 }
                  }}
                  className="lg:max-w-xl text-justify"
                >
                  {item}
                </motion.div>
              </li>
            ))}
          </ul>
        </div>
        <div>
          <Lottie options={defaultOptions} height={400} width={400} />
        </div>
      </div>
    </section>
  )
}
