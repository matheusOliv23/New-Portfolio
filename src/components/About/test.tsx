import { render } from "@testing-library/react"

import About from "./index"

describe("<About />", () => {
  it("should match snapshot", () => {
    const { container } = render(<About />)

    expect(container).toMatchSnapshot()
  })
})
