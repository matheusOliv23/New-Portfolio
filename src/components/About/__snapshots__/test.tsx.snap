// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should match snapshot 1`] = `
<div>
  <div
    class="flex w-full lg:flex-row flex-col font-openSans gap-10 max-w-7xl px-7 py-6 md:py-20 lg:py-32 m-auto xl:justify-between lg:justify-center"
  >
    <div
      class="space-y-4"
    >
      <h2
        class="text-xl text-primary"
      >
        Sobre mim
      </h2>
      <p
        class="md:text-4xl text-2xl lg:max-w-lg font-medium leading-10"
      >
        Trabalho com Desenvolvimento Web desde 2021
      </p>
      <p
        class="lg:max-w-xl text-justify"
      >
        Tenho como objetivo principal me tornar destaque no desenvolvimento web, principalmente na parte de Front-End. Sou uma pessoa muito dedicada, educada, responsável e gosto muito de novos desafios. Prezo muito pela qualidade e satisfação dos clientes e tenho a mente bem aberta para receber feedbacks, tanto positivos quanto negativos.
      </p>
    </div>
    <div
      class="flex items-center justify-center flex-col w-full lg:max-w-sm space-y-4"
    >
      <div
        class="flex w-full justify-between md:gap-16 md:justify-start lg:justify-between items-center"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-6xl font-semibold text-primary"
          >
            1.5
          </p>
          <p
            class="text-sm"
          >
            anos 
            <br />
             de experiência
          </p>
        </div>
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-6xl font-semibold text-primary"
          >
            6
          </p>
          <p
            class="text-sm"
          >
            projetos 
            <br />
             concluídos
          </p>
        </div>
      </div>
      <p
        class="text-justify"
      >
        Experiência em desenvolvimento de aplicações web responsivas utilizando Javascript, Typescript e Reactjs.
      </p>
    </div>
  </div>
</div>
`;
